2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,60 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
```

Comment:
@rhcarvalho Thank you for your effort anyway!

To fix it properly we should configure MongoDB to not print logs to stdout. And there is even option for that: [`logpath`](https://docs.mongodb.org/v2.4/reference/configuration-options/#logpath). I've tried to use `logpath = /dev/stderr` but it didn't help and logs still was printed to stdout.

Also there is a bug that sounds very similar to our problem: [--eval should be affected by the --quiet switch](https://jira.mongodb.org/browse/SERVER-4391) and it's not fixed.